pkg/estimator/server/replica/replica_test.go

```diff
@@ -0,0 +1,501 @@
+								"app": "test",
+							},
+						},
+						"template": map[string]interface{}{
+							"metadata": map[string]interface{}{
+								"labels": map[string]interface{}{
+									"app": "test",
+								},
+							},
+						},
+					},
+				},
+			},
+			threshold: 5 * time.Minute,
+			// Pod is pending and unschedulable for longer than threshold
+			pods: []*corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-pod-1",
+						Namespace: "default",
+						Labels: map[string]string{
+							"app":               "test",
+							"pod-template-hash": "xyz123",
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "apps/v1",
+								Kind:       "ReplicaSet",
+								Name:       "test-rs",
+								UID:        "test-rs-uid",
+								Controller: ptr.To[bool](true),
+							},
+						},
+					},
+					Status: corev1.PodStatus{
+						Phase: corev1.PodPending,
+						Conditions: []corev1.PodCondition{
+							{
+								Type:               corev1.PodScheduled,
+								Status:             corev1.ConditionFalse,
+								Reason:             corev1.PodReasonUnschedulable,
+								LastTransitionTime: metav1.Time{Time: time.Now().Add(-10 * time.Minute)},
+							},
+						},
+					},
+				},
+			},
+			replicaSets: []*appsv1.ReplicaSet{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-rs",
+						Namespace: "default",
+						UID:       "test-rs-uid",
+						Labels: map[string]string{
+							"app":               "test",
+							"pod-template-hash": "xyz123",
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "apps/v1",
+								Kind:       "Deployment",
+								Name:       "test-deployment",
+								UID:        "test-deployment-uid",
+								Controller: ptr.To[bool](true),
+							},
+						},
+					},
+					Spec: appsv1.ReplicaSetSpec{
+						Selector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"app":               "test",
+								"pod-template-hash": "xyz123",
+							},
+						},
+						Template: corev1.PodTemplateSpec{
+							ObjectMeta: metav1.ObjectMeta{
+								Labels: map[string]string{
+									"app":               "test",
+									"pod-template-hash": "xyz123",
+								},
+							},
+						},
+					},
+				},
+			},
+			expected:    1,
+			expectError: false,
+		},
+		{
+			name: "unsupported workload kind",
+			// Create a StatefulSet workload which is not supported
+			workload: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "apps/v1",
+					"kind":       "StatefulSet",
+					"metadata": map[string]interface{}{
+						"name":      "test-statefulset",
+						"namespace": "default",
+					},
+				},
+			},
+			threshold:   5 * time.Minute,
+			expected:    0,
+			expectError: true,
+		},
+		{
+			name: "deployment with negative threshold",
+			// Testing that negative threshold is handled correctly (should be set to 0)
```

Comment:
> // Testing that negative threshold is handled correctly (should be set to 0)

Is there a better way to test this case? The current test case doesn't seem to cover this case well.

Reply:
Good catch! I agree that the current test case for negative threshold isn't comprehensive enough. I'll improve it by adding a case that actually verifies the threshold conversion behavior.

Reply:
I saw your latest test case and it is valid to me.